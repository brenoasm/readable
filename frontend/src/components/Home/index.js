import React, { Fragment } from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';

import { colors } from '../../theme';

import CategoryList from '../CategoryList';
import PostsList from '../PostsList';
import ToolsRow from '../ToolsRow';

const propTypes = {
  posts: PropTypes.arrayOf(PropTypes.object),
  categories: PropTypes.arrayOf(PropTypes.object),
  modifyPostVoteValues: PropTypes.func,
  sortMethods: PropTypes.arrayOf(PropTypes.object),
  selectedSortMethod: PropTypes.string,
  getSelectedSortMethod: PropTypes.func,
  showModal: PropTypes.func,
  editPost: PropTypes.func,
  deletePost: PropTypes.func,
  activeRoute: PropTypes.string
};

const defaultProps = {
  posts: null,
  categories: null,
  modifyPostVoteValues: () => {},
  sortMethods: [],
  selectedSortMethod: '',
  getSelectedSortMethod: () => {},
  showModal: () => {},
  editPost: () => {},
  deletePost: () => {},
  activeRoute: ''
};

const StyledHome = styled.div`
  display: flex;
  justify-content: space-between;
  background-color: ${colors.primary.primaryFour};
  margin-top: 10px;

  > div:first-child {
    display: flex;
    flex-direction: column;
    background-color: ${colors.primary.primaryFive};
    min-width: 200px;
    box-shadow: 0 2px 5px ${colors.primary.primaryTwo};
    border-right: 1px solid ${colors.primary.primaryFive};
  }

  > div:last-child {
    min-width: calc(100% - 200px);
    max-height: 580px;
    overflow-y: auto;
    background-color: ${colors.primary.primaryFour};
    box-shadow: 0 2px 5px ${colors.primary.primaryTwo};
  }
`;

const Home = ({
  posts,
  categories,
  modifyPostVoteValues,
  sortMethods,
  selectedSortMethod,
  getSelectedSortMethod,
  showModal,
  editPost,
  deletePost,
  activeRoute
}) => (
  <Fragment>
    <ToolsRow
      showModal={showModal}
      options={sortMethods}
      selectedOption={selectedSortMethod}
      onFilterChange={getSelectedSortMethod}
    />
    <StyledHome>
      <div>
        {Array.isArray(categories) && (
          <CategoryList
            categories={categories}
            activeRoute={activeRoute}
          />
        )}
      </div>
      <div>
        {Array.isArray(posts) && (
          <PostsList
            posts={posts}
            modifyPostVoteValues={modifyPostVoteValues}
            sortMethod={selectedSortMethod}
            editPost={editPost}
            deletePost={deletePost}
          />
        )}
      </div>
    </StyledHome>
  </Fragment>
);

Home.displayName = 'Home';

Home.propTypes = propTypes;
Home.defaultProps = defaultProps;

export default Home;
