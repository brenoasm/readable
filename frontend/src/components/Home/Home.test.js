/* eslint-disable */

import React from 'react';

import Home from './';

describe('Home', () => {
  it('matches the snapshot', () => {
    const initialProps = {
      posts: [
        {
          "id": "8xf0y6ziyjabvozdd253nd",
          "timestamp": 1467166872634,
          "title": "Udacity is the best place to learn React",
          "body": "Everyone says so after all.",
          "author": "thingtwo",
          "category": "react",
          "voteScore": 6,
          "deleted": false,
          "commentCount": 2
        },
        {
          "id": "4771766477535847731044778930",
          "timestamp": 1538493219570,
          "title": "s",
          "body": "c",
          "author": "a",
          "category": "react",
          "voteScore": 1,
          "deleted": false,
          "commentCount": 0
        }
      ],
      categories: [
        {
          "name": "react",
          "path": "react"
        },
        {
          "name": "redux",
          "path": "redux"
        },
        {
          "name": "udacity",
          "path": "udacity"
        }
      ],
      modifyPostVoteValues: jest.fn(),
      sortMethods: [
        {
          "text": "Por data",
          "value": "sort_by_date"
        },
        {
          "text": "Por votos",
          "value": "sort_by_vote_score"
        }
      ],
      selectedSortMethod: 'sort_by_date',
      getSelectedSortMethod: jest.fn(),
      showModal: jest.fn(),
      editPost: jest.fn(),
      deletePost: jest.fn(),
      activeRoute: '/redux'
    };

    const wrapper = shallow(<Home {...initialProps} />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should not render a categoryList component', () => {
    const initialProps = {
      categories: null,
    };

    const wrapper = shallow(<Home {...initialProps} />);

    expect(wrapper.find('CategoryList')).toHaveLength(0);
  });

  it('should render a categoryList component', () => {
    const initialProps = {
      categories: [
        {
          "name": "react",
          "path": "react"
        },
        {
          "name": "redux",
          "path": "redux"
        },
        {
          "name": "udacity",
          "path": "udacity"
        }
      ],
    };

    const wrapper = shallow(<Home {...initialProps} />);

    expect(wrapper.find('CategoryList')).toHaveLength(1);
  });

  it('should not render a PostsList component', () => {
    const initialProps = {
      posts: null,
    };

    const wrapper = shallow(<Home {...initialProps} />);

    expect(wrapper.find('PostsList')).toHaveLength(0);
  });

  it('should render a PostsList component', () => {
    const initialProps = {
      posts: [
        {
          "id": "8xf0y6ziyjabvozdd253nd",
          "timestamp": 1467166872634,
          "title": "Udacity is the best place to learn React",
          "body": "Everyone says so after all.",
          "author": "thingtwo",
          "category": "react",
          "voteScore": 6,
          "deleted": false,
          "commentCount": 2
        },
        {
          "id": "4771766477535847731044778930",
          "timestamp": 1538493219570,
          "title": "s",
          "body": "c",
          "author": "a",
          "category": "react",
          "voteScore": 1,
          "deleted": false,
          "commentCount": 0
        }
      ],
    };

    const wrapper = shallow(<Home {...initialProps} />);

    expect(wrapper.find('PostsList')).toHaveLength(1);
  });
});
