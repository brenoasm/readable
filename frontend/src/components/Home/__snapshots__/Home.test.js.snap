// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home matches the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Home
    activeRoute="/redux"
    categories={
        Array [
            Object {
              "name": "react",
              "path": "react",
            },
            Object {
              "name": "redux",
              "path": "redux",
            },
            Object {
              "name": "udacity",
              "path": "udacity",
            },
          ]
    }
    deletePost={[Function]}
    editPost={[Function]}
    getSelectedSortMethod={[Function]}
    modifyPostVoteValues={[Function]}
    posts={
        Array [
            Object {
              "author": "thingtwo",
              "body": "Everyone says so after all.",
              "category": "react",
              "commentCount": 2,
              "deleted": false,
              "id": "8xf0y6ziyjabvozdd253nd",
              "timestamp": 1467166872634,
              "title": "Udacity is the best place to learn React",
              "voteScore": 6,
            },
            Object {
              "author": "a",
              "body": "c",
              "category": "react",
              "commentCount": 0,
              "deleted": false,
              "id": "4771766477535847731044778930",
              "timestamp": 1538493219570,
              "title": "s",
              "voteScore": 1,
            },
          ]
    }
    selectedSortMethod="sort_by_date"
    showModal={[Function]}
    sortMethods={
        Array [
            Object {
              "text": "Por data",
              "value": "sort_by_date",
            },
            Object {
              "text": "Por votos",
              "value": "sort_by_vote_score",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <ToolsRow
          onFilterChange={[Function]}
          options={
                    Array [
                              Object {
                                "text": "Por data",
                                "value": "sort_by_date",
                              },
                              Object {
                                "text": "Por votos",
                                "value": "sort_by_vote_score",
                              },
                            ]
          }
          selectedOption="sort_by_date"
          showModal={[Function]}
/>,
        <styled.div>
          <div>
                    <CategoryList
                              activeRoute="/redux"
                              categories={
                                        Array [
                                                  Object {
                                                    "name": "react",
                                                    "path": "react",
                                                  },
                                                  Object {
                                                    "name": "redux",
                                                    "path": "redux",
                                                  },
                                                  Object {
                                                    "name": "udacity",
                                                    "path": "udacity",
                                                  },
                                                ]
                              }
                    />
          </div>
          <div>
                    <PostsList
                              deletePost={[Function]}
                              editPost={[Function]}
                              modifyPostVoteValues={[Function]}
                              posts={
                                        Array [
                                                  Object {
                                                    "author": "thingtwo",
                                                    "body": "Everyone says so after all.",
                                                    "category": "react",
                                                    "commentCount": 2,
                                                    "deleted": false,
                                                    "id": "8xf0y6ziyjabvozdd253nd",
                                                    "timestamp": 1467166872634,
                                                    "title": "Udacity is the best place to learn React",
                                                    "voteScore": 6,
                                                  },
                                                  Object {
                                                    "author": "a",
                                                    "body": "c",
                                                    "category": "react",
                                                    "commentCount": 0,
                                                    "deleted": false,
                                                    "id": "4771766477535847731044778930",
                                                    "timestamp": 1538493219570,
                                                    "title": "s",
                                                    "voteScore": 1,
                                                  },
                                                ]
                              }
                              sortMethod="sort_by_date"
                    />
          </div>
</styled.div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "onFilterChange": [Function],
          "options": Array [
            Object {
              "text": "Por data",
              "value": "sort_by_date",
            },
            Object {
              "text": "Por votos",
              "value": "sort_by_vote_score",
            },
          ],
          "selectedOption": "sort_by_date",
          "showModal": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <div>
              <CategoryList
                            activeRoute="/redux"
                            categories={
                                          Array [
                                                        Object {
                                                          "name": "react",
                                                          "path": "react",
                                                        },
                                                        Object {
                                                          "name": "redux",
                                                          "path": "redux",
                                                        },
                                                        Object {
                                                          "name": "udacity",
                                                          "path": "udacity",
                                                        },
                                                      ]
                            }
              />
</div>,
            <div>
              <PostsList
                            deletePost={[Function]}
                            editPost={[Function]}
                            modifyPostVoteValues={[Function]}
                            posts={
                                          Array [
                                                        Object {
                                                          "author": "thingtwo",
                                                          "body": "Everyone says so after all.",
                                                          "category": "react",
                                                          "commentCount": 2,
                                                          "deleted": false,
                                                          "id": "8xf0y6ziyjabvozdd253nd",
                                                          "timestamp": 1467166872634,
                                                          "title": "Udacity is the best place to learn React",
                                                          "voteScore": 6,
                                                        },
                                                        Object {
                                                          "author": "a",
                                                          "body": "c",
                                                          "category": "react",
                                                          "commentCount": 0,
                                                          "deleted": false,
                                                          "id": "4771766477535847731044778930",
                                                          "timestamp": 1538493219570,
                                                          "title": "s",
                                                          "voteScore": 1,
                                                        },
                                                      ]
                            }
                            sortMethod="sort_by_date"
              />
</div>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <CategoryList
                activeRoute="/redux"
                categories={
                                Array [
                                                Object {
                                                  "name": "react",
                                                  "path": "react",
                                                },
                                                Object {
                                                  "name": "redux",
                                                  "path": "redux",
                                                },
                                                Object {
                                                  "name": "udacity",
                                                  "path": "udacity",
                                                },
                                              ]
                }
/>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeRoute": "/redux",
                "categories": Array [
                  Object {
                    "name": "react",
                    "path": "react",
                  },
                  Object {
                    "name": "redux",
                    "path": "redux",
                  },
                  Object {
                    "name": "udacity",
                    "path": "udacity",
                  },
                ],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <PostsList
                deletePost={[Function]}
                editPost={[Function]}
                modifyPostVoteValues={[Function]}
                posts={
                                Array [
                                                Object {
                                                  "author": "thingtwo",
                                                  "body": "Everyone says so after all.",
                                                  "category": "react",
                                                  "commentCount": 2,
                                                  "deleted": false,
                                                  "id": "8xf0y6ziyjabvozdd253nd",
                                                  "timestamp": 1467166872634,
                                                  "title": "Udacity is the best place to learn React",
                                                  "voteScore": 6,
                                                },
                                                Object {
                                                  "author": "a",
                                                  "body": "c",
                                                  "category": "react",
                                                  "commentCount": 0,
                                                  "deleted": false,
                                                  "id": "4771766477535847731044778930",
                                                  "timestamp": 1538493219570,
                                                  "title": "s",
                                                  "voteScore": 1,
                                                },
                                              ]
                }
                sortMethod="sort_by_date"
/>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "deletePost": [Function],
                "editPost": [Function],
                "modifyPostVoteValues": [Function],
                "posts": Array [
                  Object {
                    "author": "thingtwo",
                    "body": "Everyone says so after all.",
                    "category": "react",
                    "commentCount": 2,
                    "deleted": false,
                    "id": "8xf0y6ziyjabvozdd253nd",
                    "timestamp": 1467166872634,
                    "title": "Udacity is the best place to learn React",
                    "voteScore": 6,
                  },
                  Object {
                    "author": "a",
                    "body": "c",
                    "category": "react",
                    "commentCount": 0,
                    "deleted": false,
                    "id": "4771766477535847731044778930",
                    "timestamp": 1538493219570,
                    "title": "s",
                    "voteScore": 1,
                  },
                ],
                "sortMethod": "sort_by_date",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
        ],
        "type": [Function],
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <ToolsRow
            onFilterChange={[Function]}
            options={
                        Array [
                                    Object {
                                      "text": "Por data",
                                      "value": "sort_by_date",
                                    },
                                    Object {
                                      "text": "Por votos",
                                      "value": "sort_by_vote_score",
                                    },
                                  ]
            }
            selectedOption="sort_by_date"
            showModal={[Function]}
/>,
          <styled.div>
            <div>
                        <CategoryList
                                    activeRoute="/redux"
                                    categories={
                                                Array [
                                                            Object {
                                                              "name": "react",
                                                              "path": "react",
                                                            },
                                                            Object {
                                                              "name": "redux",
                                                              "path": "redux",
                                                            },
                                                            Object {
                                                              "name": "udacity",
                                                              "path": "udacity",
                                                            },
                                                          ]
                                    }
                        />
            </div>
            <div>
                        <PostsList
                                    deletePost={[Function]}
                                    editPost={[Function]}
                                    modifyPostVoteValues={[Function]}
                                    posts={
                                                Array [
                                                            Object {
                                                              "author": "thingtwo",
                                                              "body": "Everyone says so after all.",
                                                              "category": "react",
                                                              "commentCount": 2,
                                                              "deleted": false,
                                                              "id": "8xf0y6ziyjabvozdd253nd",
                                                              "timestamp": 1467166872634,
                                                              "title": "Udacity is the best place to learn React",
                                                              "voteScore": 6,
                                                            },
                                                            Object {
                                                              "author": "a",
                                                              "body": "c",
                                                              "category": "react",
                                                              "commentCount": 0,
                                                              "deleted": false,
                                                              "id": "4771766477535847731044778930",
                                                              "timestamp": 1538493219570,
                                                              "title": "s",
                                                              "voteScore": 1,
                                                            },
                                                          ]
                                    }
                                    sortMethod="sort_by_date"
                        />
            </div>
</styled.div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "onFilterChange": [Function],
            "options": Array [
              Object {
                "text": "Por data",
                "value": "sort_by_date",
              },
              Object {
                "text": "Por votos",
                "value": "sort_by_vote_score",
              },
            ],
            "selectedOption": "sort_by_date",
            "showModal": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <div>
                <CategoryList
                                activeRoute="/redux"
                                categories={
                                                Array [
                                                                Object {
                                                                  "name": "react",
                                                                  "path": "react",
                                                                },
                                                                Object {
                                                                  "name": "redux",
                                                                  "path": "redux",
                                                                },
                                                                Object {
                                                                  "name": "udacity",
                                                                  "path": "udacity",
                                                                },
                                                              ]
                                }
                />
</div>,
              <div>
                <PostsList
                                deletePost={[Function]}
                                editPost={[Function]}
                                modifyPostVoteValues={[Function]}
                                posts={
                                                Array [
                                                                Object {
                                                                  "author": "thingtwo",
                                                                  "body": "Everyone says so after all.",
                                                                  "category": "react",
                                                                  "commentCount": 2,
                                                                  "deleted": false,
                                                                  "id": "8xf0y6ziyjabvozdd253nd",
                                                                  "timestamp": 1467166872634,
                                                                  "title": "Udacity is the best place to learn React",
                                                                  "voteScore": 6,
                                                                },
                                                                Object {
                                                                  "author": "a",
                                                                  "body": "c",
                                                                  "category": "react",
                                                                  "commentCount": 0,
                                                                  "deleted": false,
                                                                  "id": "4771766477535847731044778930",
                                                                  "timestamp": 1538493219570,
                                                                  "title": "s",
                                                                  "voteScore": 1,
                                                                },
                                                              ]
                                }
                                sortMethod="sort_by_date"
                />
</div>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <CategoryList
                  activeRoute="/redux"
                  categories={
                                    Array [
                                                      Object {
                                                        "name": "react",
                                                        "path": "react",
                                                      },
                                                      Object {
                                                        "name": "redux",
                                                        "path": "redux",
                                                      },
                                                      Object {
                                                        "name": "udacity",
                                                        "path": "udacity",
                                                      },
                                                    ]
                  }
/>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "activeRoute": "/redux",
                  "categories": Array [
                    Object {
                      "name": "react",
                      "path": "react",
                    },
                    Object {
                      "name": "redux",
                      "path": "redux",
                    },
                    Object {
                      "name": "udacity",
                      "path": "udacity",
                    },
                  ],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <PostsList
                  deletePost={[Function]}
                  editPost={[Function]}
                  modifyPostVoteValues={[Function]}
                  posts={
                                    Array [
                                                      Object {
                                                        "author": "thingtwo",
                                                        "body": "Everyone says so after all.",
                                                        "category": "react",
                                                        "commentCount": 2,
                                                        "deleted": false,
                                                        "id": "8xf0y6ziyjabvozdd253nd",
                                                        "timestamp": 1467166872634,
                                                        "title": "Udacity is the best place to learn React",
                                                        "voteScore": 6,
                                                      },
                                                      Object {
                                                        "author": "a",
                                                        "body": "c",
                                                        "category": "react",
                                                        "commentCount": 0,
                                                        "deleted": false,
                                                        "id": "4771766477535847731044778930",
                                                        "timestamp": 1538493219570,
                                                        "title": "s",
                                                        "voteScore": 1,
                                                      },
                                                    ]
                  }
                  sortMethod="sort_by_date"
/>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "deletePost": [Function],
                  "editPost": [Function],
                  "modifyPostVoteValues": [Function],
                  "posts": Array [
                    Object {
                      "author": "thingtwo",
                      "body": "Everyone says so after all.",
                      "category": "react",
                      "commentCount": 2,
                      "deleted": false,
                      "id": "8xf0y6ziyjabvozdd253nd",
                      "timestamp": 1467166872634,
                      "title": "Udacity is the best place to learn React",
                      "voteScore": 6,
                    },
                    Object {
                      "author": "a",
                      "body": "c",
                      "category": "react",
                      "commentCount": 0,
                      "deleted": false,
                      "id": "4771766477535847731044778930",
                      "timestamp": 1538493219570,
                      "title": "s",
                      "voteScore": 1,
                    },
                  ],
                  "sortMethod": "sort_by_date",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "div",
            },
          ],
          "type": [Function],
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
